python/cuspatial/cuspatial/tests/binpreds/binpred_test_dispatch.py

```diff
@@ -0,0 +1,244 @@
+# Copyright (c) 2023, NVIDIA CORPORATION.
+
+import pytest
+from shapely.geometry import LineString, MultiPoint, Point, Polygon
+
+import cuspatial
+
+"""Test Dispatch"""
+
+"""Testing all combinations of possible geometry types and
+binary predicates is a complex task. This file contains the
+basic fixtures for all geometry types that are required to
+cover different possible test outcomes. This file also contains
+the dispatching system that uses each fixture to generate
+a test for each possible combination of geometry types and
+binary predicates.
+"""
+
+"""The following fixtures are used to generate tests for
+each possible combination of geometry types and binary
+predicates. The fixtures are combined in the test function
+in `test_binpreds_test_dispatch.py` to make the following
+tuple: (predicate, geotype, geotype, expected_result). The
+geotype fixtures are used to generate the first two
+elements of the tuple. The predicate fixture is used to
+generate the third element of the tuple. The expected_result
+fixture is used to generate the fourth element of the tuple.
+"""
+
+"""The collection of all possible binary predicates"""
+
+
+@pytest.fixture(
+    params=[
+        "contains_properly",
+        "geom_equals",
+        "intersects",
+        "covers",
+        "crosses",
+        "disjoint",
+        "overlaps",
+        "touches",
+        "within",
+    ]
+)
+def predicate(request):
+    return request.param
+
+
+"""The collection of all possible geometry types"""
+
+
+@pytest.fixture(
+    params=[
+        (Point, Point),
+        (Point, MultiPoint),
+        (Point, LineString),
+        (Point, Polygon),
+        (MultiPoint, Point),
+        (MultiPoint, MultiPoint),
+        (MultiPoint, LineString),
+        (MultiPoint, Polygon),
+        (LineString, Point),
+        (LineString, MultiPoint),
+        (LineString, LineString),
+        (LineString, Polygon),
+        (Polygon, Point),
+        (Polygon, MultiPoint),
+        (Polygon, LineString),
+        (Polygon, Polygon),
+    ]
+)
+def geotype_tuple(request):
+    return request.param
+
+
+"""The collection of test types. This section is dispatched based
+on the feature type. Each feature pairing has a specific set of
+comparisons that need to be performed to cover the entire test
+space. This section will be replaced with specific feature
+representations that cover all possible geometric combinations."""
+
+
+@pytest.fixture(
+    params=[
+        "single_equal",
+        "single_disjoint",
+        "triple_center_equal",
+        "triple_center_disjoint",
+        "border_overlap",
+        "interior_overlap",
+    ]
+)
+def test_type(request):
+    return request.param
+
+
+"""Named features for each feature dispatch."""
```

Comment:
Can you give a bit more detail on what basis did you choose these test examples from? I expect what you chose should be sufficient to cover most of  the corner cases for various geometry combination and binpreds.